app/src/main/java/com/movies/presentation/favourite/ui/adapter/FavouriteMovieAdapter.kt

```diff
@@ -2,32 +2,47 @@ package com.movies.presentation.favourite.ui.adapter
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
 import com.movies.common.extensions.loadImage
 import com.movies.databinding.MovieItemBinding
-import com.movies.presentation.base.adapter.movie_adapter.BaseMovieAdapter
 import com.movies.presentation.base.data.model.MovieUIModel
+import com.movies.presentation.base.diff_util.DiffUtilCallback
 
-class FavouriteMovieAdapter : BaseMovieAdapter<MovieUIModel, FavouriteMovieAdapter.MoviesViewHolder>() {
-
-    override fun onCreateViewHolder(
-        parent: ViewGroup,
-        viewType: Int
-    ): MoviesViewHolder {
-        return MoviesViewHolder(
-            MovieItemBinding.inflate(
-                LayoutInflater.from(parent.context),
-                parent,
-                false
-            )
+class FavouriteMovieAdapter :
+    ListAdapter<MovieUIModel, FavouriteMovieAdapter.MoviesViewHolder>(DiffUtilCallback()) {
+
+    private var onClickCallback: ((MovieUIModel) -> Unit)? = null
+    private var onFavouriteClick: ((MovieUIModel, Boolean) -> Unit)? = null
+
+    fun onItemClickListener(onClickCallback: (MovieUIModel) -> Unit) {
+        this.onClickCallback = onClickCallback
+    }
+
+    fun onFavouriteClickListener(onFavouriteClick: (MovieUIModel, Boolean) -> Unit) {
+        this.onFavouriteClick = onFavouriteClick
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MoviesViewHolder {
+        val binding = MovieItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
         )
+        return MoviesViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: MoviesViewHolder, position: Int) {
+        holder.bind(getItem(position), onClickCallback, onFavouriteClick)
     }
 
-    class MoviesViewHolder(val binding: MovieItemBinding) :
-        BaseMovieViewHolder<MovieUIModel>(binding) {
-        override fun onBindMovie(
+    class MoviesViewHolder(private val binding: MovieItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(
             item: MovieUIModel,
             onClickCallback: ((MovieUIModel) -> Unit)?,
-            onFavouriteClick: ((MovieUIModel, Boolean) -> Unit)?,
+            onFavouriteClick: ((MovieUIModel, Boolean) -> Unit)?
         ) {
```

Comment:
when you make this view holder class separately you dont need this lambdas inside the function, just pass view holder class constructor and in the init block write click listener or wire click in oncreateViewholder directly

Reply:
done